src/Lecture2.hs

```diff
@@ -70,7 +75,13 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt n (x:xs) 
+  | n == 0 = (Just x, xs)
+  | otherwise = 
+    let (ids, ys) = removeAt (n - 1) xs
+    in (ids, x:ys)
```

Comment:
It looks like your implementation hangs when the given index is negative and the list is infinite 🤔 ♾️

Reply:
When both of those happen at the same time? Or when either happens? But yeah I didn't treat the negative case, will fix it shortly, thanks!

Reply:
Just pinging here, though I fixed the negative problem

Reply:
I see that your code now handles negative cases 🙂 It's hard to tell if tests pass because CI failed for some mysterious reason 🤔 Probably something weird happened. I can recommend restarting the CI 🍏